src/dht.cpp

```diff
@@ -1287,11 +1292,17 @@ Dht::storageAddListener(const InfoHash& id, const Sp<Node>& node, size_t socket_
     if (l == node_listeners.end()) {
         auto vals = st->second.get(query.where.getFilter());
         if (not vals.empty()) {
-            network_engine.tellListener(node, socket_id, id, WANT4 | WANT6, makeToken(node->getAddr(), false),
-                    buckets4.findClosestNodes(id, now, TARGET_NODES), buckets6.findClosestNodes(id, now, TARGET_NODES),
-                    std::move(vals), query);
+            if (version == 1) {
+                network_engine.updateValues(node, socket_id, id, WANT4 | WANT6, makeToken(node->getAddr(), false),
+                        buckets4.findClosestNodes(id, now, TARGET_NODES), buckets6.findClosestNodes(id, now, TARGET_NODES),
+                        std::move(vals), query);
+            } else {
+                network_engine.tellListener(node, socket_id, id, WANT4 | WANT6, makeToken(node->getAddr(), false),
+                        buckets4.findClosestNodes(id, now, TARGET_NODES), buckets6.findClosestNodes(id, now, TARGET_NODES),
+                        std::move(vals), query);
+            }
```

Comment:
move this logic inside tellListener (it should be invisible to Dht except the version number)